packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1489 @@
+///
+/// Numeric formatting is optimized for debug message readability.
+class DoubleProperty extends _NumProperty<double> {
+
+  /// If specified, `unit` describes the unit for the value (e.g. px).
+  DoubleProperty(String name, double value, {
+    bool hidden: false,
+    String nullDescription,
+    String unit,
+    String tooltip,
+    Object defaultValue: kNoDefaultValue,
+  }) : super(
+    name,
+    value,
+    hidden: hidden,
+    nullDescription: nullDescription,
+    unit: unit,
+    tooltip: tooltip,
+    defaultValue: defaultValue,
+  );
+
+
+  /// Calling [computeValue] is allowed to throw an exception.
+  ///
+  /// If an exception is thrown, the value is set to null and the description
+  /// indicates what exception was thrown.
+  DoubleProperty.lazy(
+    String name,
+    ComputePropertyValueCallback<double> computeValue, {
+    bool hidden: false,
+    String nullDescription,
+    String unit,
+    String tooltip,
+    Object defaultValue: kNoDefaultValue,
+  }) : super.lazy(
+    name,
+    computeValue,
+    hidden: hidden,
+    nullDescription: nullDescription,
+    unit: unit,
+    tooltip: tooltip,
+    defaultValue: defaultValue,
+  );
+
+  @override
+  String numberToString() => object?.toStringAsFixed(1);
+}
+
+/// An int valued property with an optional unit the value is measured in.
+///
+/// Examples of units include 'px' and 'ms'.
+class IntProperty extends _NumProperty<int> {
+  IntProperty(String name, int value, {
+    String nullDescription,
+    bool showName: true,
+    String unit,
+    Object defaultValue: kNoDefaultValue,
+    bool hidden: false,
+  }) : super(
+    name,
+    value,
+    nullDescription: nullDescription,
+    showName: showName,
+    unit: unit,
+    defaultValue: defaultValue,
+    hidden: hidden,
+  );
+
+  @override
+  String numberToString() => object.toString();
+}
+
+/// Property which clamps a [double] to between 0 and 1 and formats it as a
+/// percentage.
+class PercentProperty extends DiagnosticsProperty<double> {
+  PercentProperty(String name, double fraction, {
+    String nullDescription,
+    bool showName: true,
+  }) : super(
+    name,
+    fraction,
+    nullDescription: nullDescription,
+    showName: showName,
+  );
+
+  @override
+  String propertyValueToString() {
+    if (object == null)
+      return object.toString();
+
+    return '${(object.clamp(0.0, 1.0) * 100.0).toStringAsFixed(1)}%';
+  }
+}
+
+/// Property which handles [Matrix4] that represent transforms.
+class TransformProperty extends DiagnosticsProperty<Matrix4> {
+  TransformProperty(String name, Matrix4 value, {
+    Object defaultValue: kNoDefaultValue,
+  }) : super(
+    name,
+    value,
+    defaultValue: defaultValue,
+  );
+
+  @override
+  String propertyValueToString() => debugDescribeTransform(object).join('\n');
+}
+
+/// Property with a bool value described by custom strings.
+///
+/// The property is only shown if there is a flag string matching the bool
```

Comment:
cleaned up a bit. used `[]` references to class members as appropriate and made document a little better.
Documenting this class felt a bit awkward so I do wonder if I'm just missing a better name or description for it to make all the docs flow better.